packages/vscode-extension/src/uriHandler.ts

```diff
@@ -63,6 +65,14 @@ export class UriHandler extends vscode.EventEmitter<vscode.Uri> implements vscod
             isRunning = false;
           }
         );
+    } else if (queryParamas.referrer === Referrer.OfficeDoc) {
+      if (!queryParamas.sampleId) {
+        void vscode.window.showErrorMessage(
+          localize("teamstoolkit.devPortalIntegration.invalidLink")
+        );
+        return;
+      }
+      void vscode.commands.executeCommand("fx-extension.openSamples", false, queryParamas.sampleId);
```

Comment:
You could use `ExternalUrl` instead of `false` which is the `trigger-from` property in telemetry event. Otherwise, it will be treated as command palette.

Reply:
Done